django/db/backends/sqlite3/schema.py

```diff
@@ -104,6 +104,12 @@ def is_self_referential(f):
             f.name: f.clone() if is_self_referential(f) else f
             for f in model._meta.local_concrete_fields
         }
+
+        # Copy the CompositePrimaryKey to the new table.
+        pk = model._meta.pk
+        if isinstance(pk, CompositePrimaryKey):
+            body[pk.name] = pk.clone()
```

Comment:
It's because only the `local_concrete_fields` fields are being cloned. This wasn't an issue before because primary keys were always concrete. `CompositePrimaryKey` isn't concrete (`column is None`), but it does affect the schema.